scripts/manifests_to_csv.py

```diff
@@ -0,0 +1,73 @@
+#! /usr/bin/env python
+
+# Stand-alone python script to generate a CSV file of IIIF manifest urls
+# and shelfmarks, and optional view urls in a format suitable
+# for import into PGP application using the add_fragment_urls manage command.
+#
+# Script requires a source directory of IIIF manifest files (currently
+# assumes all json files in that directory are manifests) and a filename
+# where the resulting csv file should be stored.  View urls are
+# extracted from rendering attribute on the manifest, if present.
+#
+# Example use:
+#
+#   python scripts/manifests_to_csv.py -s manifests_dir -o iiif_urls.csv
+#
+
+import argparse
+import csv
+import glob
+import json
+import os.path
+
+
+def parse_manifests(source_dir, outfile):
+    with open(outfile, "w") as f:
+        csvwriter = csv.writer(f)
+        csvwriter.writerow(["shelfmark", "iiif_url", "url", "orig_shelfmark"])
+
+        for manifest_file in glob.iglob(os.path.join(source_dir, "*.json")):
+            with open(manifest_file) as mfile:
+                manifest = json.load(mfile)
+                iiif_url = manifest["@id"]
+                # for bodleian manifests, we're putting it in the label;
+                # in cudl it's in the metadata as a classmark (handle later if needed)
+                shelfmark = manifest["label"]
+                view_url = ""
+                if "rendering" in manifest:
+                    # NOTE: could be multiple; assume simple case for now
+                    view_url = manifest["rendering"]["@id"]
```

Comment:
yes — they do have a public view, and I figured out how to generate the link from the ids in the TEI; I'm putting this in the csv so that it will get imported as the fragment url, which we use in the image rights to link the shelfmark to the library or museum's view of that object